### ISSUE TITLE
Exceptions thrown from within custom LocalizableString implementation used in analyzer not resulting in info diagnostic during command line builds

### ISSUE THREAD

**Issue by shyamnamboodiripad** [dotnet/roslyn#1759]
1. Create an analyzer using template
2. Create a custom derived type of LocalizableString that throws an exception from within its GetText() method.
3. Replace the initializer for MessageFormat in the analyzer with code to instantiate the type created in 2 above ```internal static readonly LocalizableString MessageFormat = new MyLocalizableString(); ```
4. Create a console app and add the above analyzer as a nuget analyzer reference
5. Open this console app - notice that the exception from 2 above results in an info diagnostic in the error list
6. Now change build output verbosity for VS to detailed as described in the resolution for #1508 (so that info diagnostic for the above exception is displayed in the command line build output)
7. Build the solution

Expected: Info diagnostic for exception is displayed in the build output
Actual: No info diagnostic for above exception in build output

Per discussion with @mavasani, the bug may be that - *We never hook up exception handler for exceptions within localizable string in command line case. We do so for other exceptions from analyzers during explicit callbacks (such as from within Initialize(), Analyze...() etc.).*

**Comment by mavasani** [comment]
@shyamnamboodiripad I am unable to repro this, i.e. I do the see the info diagnostic in output window after changing msbuild verbosity to detailed.

Can you please verify if it still repros for you? 

Also make sure that you don't receive the warning about "no analyzers present" in the build output. Common scenario where an analyzer diagnostic shows up in error list but not command line is build is if the toolset compiler has a different version Microsoft.CodeAnalysis then the one referenced by the analyzer.

My analyzer uses the below custom LocalizableString implementation:
```
public class MyLocString : LocalizableString
    {
        protected override bool AreEqual(object other)
        {
            throw new NotImplementedException();
        }

        protected override int GetHash()
        {
            throw new NotImplementedException();
        }

        protected override string GetText(IFormatProvider formatProvider)
        {
            throw new NotImplementedException();
        }
    }
```

I see the below in error list:
```
Severity	Code	Description	Project	File	Line
Message	AD0001	The User Diagnostic Analyzer 'Analyzer1.Analyzer1Analyzer' threw an exception of type 'System.NotImplementedException' with message 'The method or operation is not implemented.'.	ClassLibrary2		1

```
I see the below in the output window:
```
1>  C:\Program Files (x86)\MSBuild\14.0\bin\csc2.exe /noconfig /nowarn:1701,1702,2008 /nostdlib+ /errorreport:prompt /warn:4 /define:DEBUG;TRACE /errorendlocation /preferreduilang:en-US /highentropyva+ /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\Microsoft.CSharp.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\mscorlib.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.Core.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.Data.DataSetExtensions.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.Data.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.Net.Http.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.Xml.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.2\System.Xml.Linq.dll" /debug+ /debug:full /filealign:512 /optimize- /out:obj\Debug\ClassLibrary2.dll /ruleset:"C:\Program Files (x86)\Microsoft Visual Studio 14.0\Team Tools\Static Analysis Tools\\Rule Sets\MinimumRecommendedRules.ruleset" /subsystemversion:6.00 /target:library /utf8output /analyzer:..\ClassLibrary1\bin\Debug\ClassLibrary1.dll Class1.cs Properties\AssemblyInfo.cs "C:\Users\mavasani\AppData\Local\Temp\.NETFramework,Version=v4.5.2.AssemblyAttributes.cs" obj\Debug\\TemporaryGeneratedFile_E7A71F73-0F8D-4B9B-B56E-8E70B10BC5D3.cs obj\Debug\\TemporaryGeneratedFile_036C0B5B-1481-4323-8D20-8F5ADCB23D92.cs obj\Debug\\TemporaryGeneratedFile_5937a670-0e60-4077-877b-f7221da3dda1.cs
1>  Microsoft (R) Visual C# Compiler version 1.0.0.50420
1>  Copyright (C) Microsoft Corporation. All rights reserved.
1>  info AD0001: The Compiler Analyzer 'Analyzer1.Analyzer1Analyzer' threw an exception of type 'System.NotImplementedException' with message 'The method or operation is not implemented.'.
```

**Comment by shyamnamboodiripad** [comment]
That is strange here is what I see on a vanilla install of VS build from 5/11 (so there should be no difference in version of MS.CA.dll in play)

```
1>Using "Csc" task from assembly "Microsoft.Build.Tasks.Roslyn, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35".
1>Task "Csc"
1>  C:\Program Files (x86)\MSBuild\14.0\bin\csc2.exe *** /analyzer:..\packages\Analyzer1.1.0.0.0\tools\analyzers\Analyzer1.dll ***
1>  Microsoft (R) Visual C# Compiler version 1.0.0.50510
1>  Copyright (C) Microsoft Corporation. All rights reserved.
1>C:\Users\**\Documents\Visual Studio 2015\Projects\ConsoleApplication2\ConsoleApplication2\Program.cs(9,11,9,18): warning Analyzer1: 
1>Done executing task "Csc".
```

Notice the empty warning above. But there is no info diagnostic about the crash.

I do see the crash in error list -
![capture](https://cloud.githubusercontent.com/assets/10579684/7668738/402a0ab0-fbfd-11e4-8a34-0a7bb45d7a92.PNG)

Are you sure you set the 'MessageFormat' field in the template generated analyzer to the throwing localizable string (in other words did you follow the above repro steps verbatim)?

**Comment by mavasani** [comment]
@shyamnamboodiripad I am able to reproduce this now, but weird that I did see the exception diagnostic in output window earlier.
The issue seems to be that we report analyzer + analyzer exception diagnostics to together. However, computing the message for analyzer diagnostic might throw during reporting these errors, but we don't take a look again at the original diagnostic bag for exception diagnostics. I will fix this later this week, and move it to 1.1 as this doesn't seem too critical a scenario.

**Comment by mavasani** [comment]
@shyamnamboodiripad Can you please verify the fix?
